icing/icing_optimisationProofsScript.sml

```diff
@@ -603,11 +603,376 @@ Proof
   \\ fs[]
 QED
 
+fun rename_each [] = ALL_TAC
+| rename_each [first] = qmatch_asmsub_rename_tac first \\ qpat_x_assum first mp_tac \\ disch_then assume_tac
+| rename_each (first::rest) = qmatch_asmsub_rename_tac first \\ qpat_x_assum first mp_tac \\ rename_each rest \\ disch_then assume_tac
```

Comment:
`disch_then assume_tac` -> `strip_tac` for brevity

Reply:
In this case, they are probably equivalent (because what is being stripped was originally an assumption), but note that `disch_then assume_tac` and `strip_tac` are not always equivalent.  In particular, they behave differently on goals such as `(?x. P x) ==> Q`, `p /\ q ==> r` and `p \/ q ==> r`.

Reply:
Thanks for pointing this out!